File: Backend/EduQuizWebAPI/Models/QuizModel.cs
using EduQuizDBAccess.Entities;

namespace EduQuizWebAPI.Models {
    public class QuizModel {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string Description { get; set; } = null!;
        public DateTime CreationDate { get; set; }
        public ICollection<QuestionModel>? Questions { get; set; }
        public QuizSetting? Settings { get; set; }
    }
}

File: Frontend/edu-quiz/src/app/models/quiz.model.ts
import { Question } from "./question.model";
import { QuizSettings } from "./quiz-settings.model";

export interface QuizModel {
    id: number,
    userId: number,
    name: string,
    description: string,
    creationDate: string,
    questions: Question[],
    settings: QuizSettings
}